src/app/templates/query_result.html

```diff
@@ -7,60 +7,166 @@ <h1>No QUERY RAN</h1>
+                        </td>
+                    {% endfor %}
+                </tr>
             {% endfor %}
         </tbody>
     </table>
 </div>
+
+<script type="text/javascript">
+    // Injected data from Flask (Jinja) into JavaScript as a JSON object
+    const data = JSON.parse('{{ data | tojson }}');
+
+    document.getElementById('download-btn').addEventListener('click', function () {
+            // Convert the data to a JSON string
+            const jsonString = JSON.stringify(data, null, 2);  // 'null, 2' formats it with indentation
+
+            // Create a blob with the JSON content and set the MIME type to application/json
+            const blob = new Blob([jsonString], { type: 'application/json' });
+
+            // Create a temporary link element
+            const a = document.createElement('a');
+            a.href = URL.createObjectURL(blob);
+            a.download = 'results.json';  // Set the file name
+            document.body.appendChild(a);  // Append the link to the body (it won't be visible)
+            a.click();  // Trigger the download by simulating a click
+            document.body.removeChild(a);  // Clean up and remove the link element
+        });
+
+    // Add an event listener to each download link
+    document.querySelectorAll('.download-link').forEach(link => {
+        link.addEventListener('click', function (event) {
+            event.preventDefault();  // Prevent the default link behavior
+
+            const rawData = this.getAttribute('data-rawdata');  // Get the base64-encoded data
+            const columnName = this.getAttribute('data-column');  // Get the column name
+
+            // Make a POST request to the /download_raw endpoint to initiate the download
+            fetch(`/download_raw/${columnName}`, {
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                },
+                body: JSON.stringify({ raw_data: rawData })  // Send the raw data to the backend
+            })
+            .then(response => {
+                if (response.ok) {
+                    return response.blob();  // Convert the response to a blob (the file)
+                }
+                throw new Error('Network response was not ok.');
+            })
+            .then(blob => {
+                // Create a temporary link element to download the file
+                const url = window.URL.createObjectURL(blob);
+                const a = document.createElement('a');
+                a.style.display = 'none';
+                a.href = url;
+                a.download = `${columnName}.parquet`;  // Set the file name
+                document.body.appendChild(a);
+                a.click();
+                window.URL.revokeObjectURL(url);  // Clean up the object URL
+            })
+            .catch(error => {
+                console.error('Error during file download:', error);
+            });
+        });
+    });
```

Comment:
All of this is redundant from when it was to download the info for that specific column. Should be deleted.